### PR TITLE
Schema Registry: Update support for 415 error code. Regenerate test playback files.

### PR DESCRIPTION
**Pull request by conniey**
# Description

* Regenerates the autorest API to support an explicit 415.
* Swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/16738

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../../CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/schemaregistry/azure-data-schemaregistry/CHANGELOG.md [hunk 1]
```diff
@@ -11,9 +11,13 @@
 
 ### Bugs Fixed
 
+- Added correct User Agent string for client.
+
 ### Other Changes
 
 - Regenerated REST API based off 2021-10 swagger.
+- An HttpResponseException with status code 415 is returned if an invalid SchemaFormat is passed for registerSchema or
```
Comment (lmolkova) [inline sdk/schemaregistry/azure-data-schemaregistry/CHANGELOG.md:19]:
```suggestion
- An HttpResponseException with status code 415 is returned if an invalid SchemaFormat is passed for registerSchema or
```

File: sdk/schemaregistry/azure-data-schemaregistry/CHANGELOG.md [hunk 2]
```diff
@@ -11,9 +11,13 @@
...
+  getSchemaById calls.
```
Comment (lmolkova) [inline sdk/schemaregistry/azure-data-schemaregistry/CHANGELOG.md:20]:
```suggestion
- An `HttpResponseException` with status code 415 is returned if an invalid `SchemaFormat` is passed for `registerSchema` or `getSchemaById` calls.
```

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/implementation/SchemasImpl.java [hunk 3]
```diff
@@ -75,26 +75,34 @@
...
+                value = ErrorException.class,
+                code = {415})
         @UnexpectedResponseExceptionType(ErrorException.class)
         Mono<SchemasQueryIdByContentResponse> queryIdByContent(
                 @HostParam("endpoint") String endpoint,
                 @PathParam("groupName") String groupName,
                 @PathParam("schemaName") String schemaName,
                 @QueryParam("api-version") String apiVersion,
-                @BodyParam("application/json; serialization=avro") String schemaContent,
+                @BodyParam("application/json; serialization=Avro") String schemaContent,
```
Comment (lmolkova) [inline sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/implementation/SchemasImpl.java:88]: for my own education: if this is eventually ends up in content-type header, why does the case (`avro` -> `Avro`) matter?

**Review by lmolkova — APPROVED** [review]

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/implementation/SchemasImpl.java (hunk 3 above)
Comment (conniey) [reply sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/implementation/SchemasImpl.java:88]: It doesn't. I can remove it to reduce the diff

**Pull request merged**